pkg/kgen/kgen.go

```diff
@@ -36,6 +39,13 @@ func Generate(opts GenerateOpts) *v1.Pod {
 		},
 	}
 
+	if opts.Namespace == "" {
+		pod.ObjectMeta.Namespace = opts.Namespace
+	}
+	if len(opts.Namespace) != 0 {
+		pod.ObjectMeta.Namespace = opts.Namespace
+	}
```

Comment:
```suggestion
	pod.ObjectMeta.Namespace = opts.Namespace
```
I would just simplify with the following :D!